app/src/0_simple_agent.py

```diff
@@ -2,107 +2,137 @@
-    When you are asked to search the web you must use a tool.
-    """ ,
-    tools=["mcp::custom_tool", torchtune],
-    tool_config={"tool_choice":"auto"}
-)
-
-if args.auto:
-    user_prompts = ["""please execute the client tool torchtune to answer the following question: what is torchtune?""",
-                    """give me a random number between 1 and 1010"""]
-    session_id = agent.create_session(session_name="Auto_demo")
-    for prompt in user_prompts:
-        turn_response = agent.create_turn(
-            messages=[
-                {
-                    "role":"user",
-                    "content": prompt
-                }
-            ],
-            session_id=session_id,
-            stream=True,
-        )
-        for log in EventLogger().log(turn_response):
-            log.print()
-
-else:
-    #Create a chat session
-    session_id = agent.create_session(session_name="Conversation_demo")
-    while True:
-        user_input = input(">>> ") 
-        if "/bye" in user_input:
-            if args.session_info_on_exit:
-                agent_session = client.agents.session.retrieve(session_id=session_id, agent_id=agent.agent_id)
-                print( agent_session.to_dict())
-            break
-        turn_response = agent.create_turn(
-            session_id=session_id,
-            messages=[{"role": "user", "content": user_input}],
-        )
-
-        for log in EventLogger().log(turn_response):
-            log.print()
-
+import json
+
+
+
+
+def agent_prompt(file_path,REMOTE_MCP_URL, registation_name):
+
+    print(f"Testing {registation_name} toolgroup ")
+    # Open the JSON file and load its content
+    with open(file_path, 'r') as file:
+        data = json.load(file)
+
+    queries = data['queries']
+    user_prompts = [query['query'] for query in queries]
+    model = data['meta']['model']
+    print(f'Model is {model} \n \n')
+
+
+    load_dotenv()
+
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(message)s')
+    stream_handler.setFormatter(formatter)
+    logger.addHandler(stream_handler)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-r", "--remote", help="Uses the remote_url", action="store_true")
+    parser.add_argument("-s", "--session-info-on-exit", help="Prints agent session info on exit", action="store_true")
+    parser.add_argument("-a", "--auto", help="Automatically runs examples, and does not start a chat session", action="store_true")
+    args = parser.parse_args()
+
+
+    # Connect to a llama stack server
+    if args.remote:
+        base_url = os.getenv('REMOTE_BASE_URL')
+        print(f"base_url is {base_url}")
+        mcp_url = REMOTE_MCP_URL
+    else:
+        base_url="http://localhost:8321"
+        mcp_url="http://host.containers.internal:8000/sse"
```

Comment:
Let's just use the environment variables here and delete whats in the `else` clause.